app/app.go

```diff
@@ -690,6 +683,18 @@ func NewGaiaApp(
 				FeegrantKeeper:  app.FeeGrantKeeper,
 				SignModeHandler: encodingConfig.TxConfig.SignModeHandler(),
 				SigGasConsumer:  ante.DefaultSigVerificationGasConsumer,
+				// TxFeeChecker is not the default fee check, it will not check if the fee meets min_gas_price, this is checked in NewFeeWithBypassDecorator already.
+				TxFeeChecker: func(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
+					feeTx, ok := tx.(sdk.FeeTx)
+					if !ok {
+						return nil, 0, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+					}
+
+					feeCoins := feeTx.GetFee()
+					priority := gaiaante.GetTxPriority(feeCoins)
+
+					return feeCoins, priority, nil
+				},
 			},
 			IBCkeeper:            app.IBCKeeper,
 			BypassMinFeeMsgTypes: cast.ToStringSlice(appOpts.Get(gaiaappparams.BypassMinFeeMsgTypesKey)),
```

Comment:
OK i did some digging to figure out what exactly the `Get` is doing here.
it's an interface referenced [here](https://github.dev/cosmos/cosmos-sdk/blob/af8ad3d20d5c78b730063c8ed8d42f8ee8d22232/server/types/app.go#L34-L35).
The AppOptions are created [here](https://github.dev/cosmos/cosmos-sdk/blob/38a11320247c46c4454c345c67c6b9a73a548cca/server/start.go#L274) by passing `ctx.Viper` which means the `Get` method is defined by Viper [here](https://github.dev/spf13/viper/blob/da55858fffe6093e995244bfc0aa2d9fba370e30/viper.go#L884) and then cast with the `ToStringSlice` defined [here](https://github.dev/spf13/cast/blob/2b0eb0f724e320b655240e331aef36d1175986c2/caste.go#L1234).


It makes me believe that if the key was not present in the config file, `Get` would return `nil` whereas if it was defined but left empty (as in `""`) that it would return as an empty string.

That means that in this line of `app.go` we could separate the command so that it was:
```
bypassMinFeeConfig := appOpts.Get(gaiaappparams.BypassMinFeeMsgTypesKey)
if bypassMinFeeConfig != nil {
  bypassMinFeeMsgTypes := cast.ToStringSlice(appOpts.Get(gaiaappparams.BypassMinFeeMsgTypesKey))
} else {
  bypassMinFeeMsgTypes := " our list of default passed messages"
}
```

This way when people upgrae they don't have to add a new line to their config file to start using the new feature (and this PR will pass the E2E test that uses default Cosmos SDK config generation.

Reply:
would be great if @alexanderbez could confirm this would work as expected

Reply:
@yaruwangway would it be possible to test the code snippet billy mentioned via a unit test?

Reply:
So your understanding on how AppOpts work is correct @okwme, however, regarding defaults, this is what happens:

1. Gaia defines the defaults [here](https://github.com/cosmos/cosmos-sdk/issues/12244#issuecomment-1155350896) when `app.toml` does NOT exist. Note, I don't understand why the default was commented out.
2. The Gaia config extension is defined [here](https://github.com/cosmos/gaia/blob/main/app/params/config.go)
3. When `app.toml` DOES exist, it will parse an empty `[]string` for `bypass-min-fee-msg-types`.

So regardless if the result is `nil` or `[]string`, the ante-handler should still work fine as iterating over a `nil` slice is a no-op.

Reply:
> It makes me believe that if the key was not present in the config file, Get would return nil whereas if it was defined but left empty (as in "") that it would return as an empty string.

No, it will return `nil` or `[]string{...}`.

Reply:
@okwme, do we want all nodes have the same bypass tx types ? i thought this is depending on what each node's wish. which means if you want to bypass ibc tx, you need to find a node that setup ibc as bypass msg types.

Reply:
@be

> So your understanding on how AppOpts work is correct @okwme, however, regarding defaults, this is what happens:
> 
> 1. Gaia defines the defaults [here](https://github.com/cosmos/cosmos-sdk/issues/12244#issuecomment-1155350896) when `app.toml` does NOT exist. Note, I don't understand why the default was commented out???
> 2. The Gaia config extension is defined [here](https://github.com/cosmos/gaia/blob/main/app/params/config.go)
> 3. When `app.toml` DOES exist, it will parse an empty `[]string` for `bypass-min-fee-msg-types` when not explicitly set.
> 
> So regardless if the result is `nil` or `[]string`, the ante-handler should still work fine as iterating over a `nil` slice is a no-op.

Hi @alexanderbez, can you check the the above link about the "Gaia defines the defaults here" ? I will have a look.

Reply:
Whoops! So sorry!

Here is the intended link: https://github.com/cosmos/gaia/blob/main/cmd/gaiad/cmd/root.go#L105-L109

Reply:
thanks @alexanderbez !
you mentioned that if no app.go exists the result will be an empty array of strings but what happens if the app.go exsists however the value for the bypassminfee messages is not set?
I made this PR to demonstrate what I thought would be a solution to ensure node operators who don't update their config will use the default set of messages. Can you confirm this thinking is correct and the desired results would be achieved?

https://github.com/cosmos/gaia/pull/1567

Reply:
merged this PR!